stubmodify/modify_test.go

```diff
@@ -78,13 +78,24 @@ func TestWriteAttributionCodeFull(t *testing.T) {
 			t.Error("fileBytes was modified in WriteAttributionCode")
 		}
 
+		// Ensure the size didn't change
 		if len(modBytes) != len(fileBytes) {
 			t.Errorf("modBytes: %d != fileBytes: %d", modBytes, fileBytes)
 		}
 
+		// Ensure the attribution code is included in the modified bytes
 		if !bytes.Contains(modBytes, code) {
 			t.Errorf("modBytes does not contain code: %v", code)
 		}
+
+		// Ensure everything after the tag and code ended up zeroed out
+		nulStart := attrOffset + len(MozTag) + len(code)
+		nulEnd := attrOffset + MaxLength
+		unusedAttrSpace := modBytes[nulStart:nulEnd]
+		nulsFound := bytes.Count(unusedAttrSpace, []byte("\x00"))
+		if nulsFound != nulEnd-nulStart {
+			t.Errorf("Expected %v nuls, found %v", nulEnd-nulStart, nulsFound)
+		}
```

Comment:
We could possibly add:

```suggestion
		}

		// Check the next 10 bytes after the attribution space.
		posAfterAttribution := attrOffset + MaxLength - len(MozTag)
		for i := 0; i < 10; i++ {
			pos := posAfterAttribution + i
			if fileBytes[pos] != modBytes[pos] {
				t.Errorf("At position %d, expected: 0x%02x but got: 0x%02x", pos, fileBytes[pos], modBytes[pos])
			}
		}
```

Reply:
Only checking a few bytes afterwards. This should currently fail with the current patch.